### PR TITLE
Add the migration to make this work with DJ 1.8

### PR DESCRIPTION
**Pull request by jimgraham**
This is complicated. We backported changes Django Oauth Toolkit (DOT) from
0.10.0, which works on DJ 1.8, to this version (0.7.1)

We missed one migration, to drop the `NOT NULL` user constraint. This is
necessary for DJ 1.8. We didn't see this because the table changes had
been applied manually.

This fix adds the migration for the user `NOT NULL` only. It is now
out-of-sync with DOT master because their `0002` migration made other
changes.

When we go back to DOT master, we will have to unwind theses changes
manually.

### CODE REVIEW

**Comment by jimgraham** [comment]
@ryanwilsonperkin @pobed2 Review please?

@ashchristopher signed off on this verbally on HipChat

screen shot 2016-05-09 at 10 42 00 am

File: oauth2_provider/migrations/0002_07_user_null.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    dependencies = [
+        ('oauth2_provider', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='AccessToken',
+            name='user',
+            field=models.ForeignKey(blank=True, to=settings.AUTH_USER_MODEL, null=True),
```
Comment (pobed2) [inline oauth2_provider/migrations/0002_07_user_null.py]: I can never remember why we need `blank=True`. Is it for form validation? And why do we need it here?

Comment (jimgraham) [inline oauth2_provider/migrations/0002_07_user_null.py]: I'm not sure. This is taken directly from https://github.com/evonove/django-oauth-toolkit/blob/000c44a252b0b4b641fa56f25a920372ceb48ace/oauth2_provider/migrations/0002_08_updates.py#L33 and I want to keep this as close to master as possible.

Comment (pobed2) [inline oauth2_provider/migrations/0002_07_user_null.py]: 👌

File: README.rst [hunk 2]
```diff
@@ -90,6 +90,16 @@ Roadmap / Todo list (help wanted)
 Changelog
 ---------
 
+0.7.1.1-wave [2016-05-07]
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Fix migration to make this Django-1.8 compatible.
+
+0.7.1-wave [2016-05-07]
```
Comment (ryanwilsonperkin) [inline README.rst]: Add a `(DEPRECATED - Do Not Use)` or equivalent suffix on this line.

File: oauth2_provider/migrations/0002_07_user_null.py (hunk 1 above)
Comment (ryanwilsonperkin) [inline oauth2_provider/migrations/0002_07_user_null.py]: Yeah, `blank=True` will allow no-value to be selected in a form.
https://docs.djangoproject.com/en/1.7/ref/models/fields/#blank

File: README.rst [hunk 3]
```diff
@@ -90,6 +90,17 @@ Roadmap / Todo list (help wanted)
 Changelog
 ---------
 
+0.7.1.1-wave [2016-05-07]
```
Comment (ryanwilsonperkin) [inline README.rst]: Update date to today

**Comment by pobed2** [comment]
LGTM

**Comment by ryanwilsonperkin** [comment]
👍

**Pull request merged**